### PR TITLE
Add MPAS App to Chiltepin

### LINKED ISSUE #96: MPAS-UW integration
We need to build a federated MPAS App.  This will require integrating MPAS UW drivers into Chiltepin as well as building Parsl Apps to install all necessary software and run the needed utilities.  A user should be able to run an MPAS workflow both locally and remotely.

The initial design of these Parsl Apps could follow the example given by the [implementation of the JEDI QG model](https://github.com/NOAA-GSL/ExascaleWorkflowSandbox/blob/main/src/chiltepin/jedi/qg/wrapper.py) used for development and testing.  It is anticipated, but not required, that the UW-Tools API will be used rather than its CLI to keep the interface in native Python for clarity and simplicity.

It is preferable to build the MPAS App to be completely self-contained such that it does not depend on preexisting data or software on a given machine.  To accomplish this, we will need Apps that do the following:

1. Fetch the MPAS static files for mesh generation
2. Install the [MPAS-Limited-Area](https://github.com/MPAS-Dev/MPAS-Limited-Area) utility
3. Use MPAS-Limited-Area to generate conus mesh files for a given horizontal resolution
4. Fetch and build the [Metis](https://github.com/KarypisLab/METIS) software to obtain the `gpmetis` executable
5. Run `gpmetis` to obtain the necessary partitioning of the mesh files for running mpas executables
6. Fetch the GFS files used for initial and boundary conditions
7. Fetch and build the [WPS](https://github.com/wrf-model/WPS) software to obtain the `ungrib.exe` executable
8. Use UWTools to run `ungrib.exe` to produce ics and lbcs
9. Fetch and build the [MPAS](https://github.com/MPAS-Dev/MPAS-Model) code
10. Use UWTools to run `init_atmosphere_model` to create ics and lbcs for model input
11. Use UWTools to run `atmosphere_model` to run a forecast

The integration of UW drivers via the uw-tools API will require adding the uw-tools Python package to our software stack, both in the container and in the `./install.sh` scripts for local HPC systems.  To minimize incompatibility problems and additional complexity across both HPC and containerized Spack environments, the uw-tools should first be tested using a pip installation rather than the conda-based install.  This should be as simple as `pip install uw-tools`, but the command must be run with the `pip` that is installed in the Spack environment.

Finally, once completed, a CI workflow should be added to test the MPAS app for a low resolution (e.g. 480km) run.  This CI file needs to run the MPAS app in the container.  See [here](https://github.com/NOAA-GSL/ExascaleWorkflowSandbox/blob/0072deffa724e14898d7e55331f79ab59940bb09/.github/workflows/docker-slurm.yml#L355) for an example of how to start the container and run commands in it.  Note that the CI workflow for running the MPAS App need not build or push any container images.  It only needs to "up" the container and issue commands to it with `docker exec ...`

This issue should be updated as futher details and plans emerge.

### PR DESCRIPTION
**Pull request by christopherwharrop-noaa**
This PR adds a self-contained MPAS app to demonstrate Chiltepin workflows for a "real" NWP model.  The App is self-contained such that, given a proper spack-stack or other supporting stack, it will download, build, and install all necessary modeling components, fetch input data and generate grid mesh files, and run all steps from ingesting the GFS grib files to the forecast.  Closes #96

### CODE REVIEW

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - This is ready for your review

**Review by NaureenBharwaniNOAA — COMMENTED** [review]
Here's my initial review.

File: requirements.txt [hunk 1]
```diff
@@ -2,22 +2,21 @@ pytest
...
 # if uwtools is installed with pip while a conda
 # environment (including base) is activated, pip
 # will not install any uwtools dependencies.
 f90nml==1.4.*
-iotaa==0.8.*
+iotaa==0.7.*
 jinja2==3.1.*
-jsonschema==4.22.*
-lxml==5.2.*
+jsonschema==4.21.*
```
[the remark this answers is not in the document]
Comment (christopherwharrop-noaa) [reply requirements.txt:20]: Yes.  This matches uwtools requirements.

File: src/chiltepin/utils/data_locations.yml
[the remark this answers is not in the document]
Comment (christopherwharrop-noaa) [reply src/chiltepin/utils/data_locations.yml:1]: It is copied verbatim from the mpas_app.  And I'm not going to change it unless the mpas app changes it.

File: src/chiltepin/mpas/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        clone_executors=["service"],
+        make_executors=["service"],
+    ):
+        def install(
+            jobs=8,
+            stdout=None,
+            stderr=None,
+        ):
+            clone_task = self.get_clone_task(executors=clone_executors)
+            make_task = self.get_make_task(executors=make_executors)
```
[the remark this answers is not in the document]
Comment (christopherwharrop-noaa) [reply src/chiltepin/mpas/wrapper.py:103]: That won't work and isn't appropriate here.

File: apps/mpas/config/user_config.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+user:
+  experiment_dir: /home/admin/work/apps/mpas/experiments/test
+  platform: chiltepin
+  resolution: 240
```
[the remark this answers is not in the document]
Comment (christopherwharrop-noaa) [reply apps/mpas/config/user_config.yaml]: i just changed it to 480 to make it run as fast as possible. That is the coarsest possible resolution.

File: src/chiltepin/utils/retrieve_data.py
[the remark this answers is not in the document]
Comment (christopherwharrop-noaa) [reply src/chiltepin/utils/retrieve_data.py:1]: This code is copied verbatim from mpas_app.  If they change it, I'll change it.

**Comment by christopherwharrop-noaa** [comment]
@NaureenBharwaniNOAA - I changed my mind about the data retrieval documentation.  I added some notes at the top of the python code and the corresponding config file to explain its provenance.  However, I did not alter the original documentation inside.  Hopefully that helps to address your comments on those.

**Review by NaureenBharwaniNOAA — APPROVED** [review]
I really appreciate the cleanup from `chiltepin` and `ci` to `docker`. It helps clarify things quite a bit. LGTM!

**Pull request merged**